src/Compilers/Core/Portable/VersionHelper.cs

```diff
@@ -74,35 +74,50 @@ private static bool TryParse(string s, bool allowWildcard, ushort maxValue, out
 
             if (hasWildcard)
             {
-                int seconds = ((int)(DateTime.Now.TimeOfDay.TotalSeconds)) / 2;
-                if (seconds > (int)maxValue)
+                for (int i = lastExplicitValue; i < values.Length; i++)
                 {
-                    version = null;
-                    return false;
-                }
-                values[3] = (ushort)seconds;
-
-                if (elements.Length == 3)
-                {
-                    TimeSpan days = DateTime.Today - new DateTime(2000, 1, 1);
-                    int build = (int)days.TotalDays;
-
-                    if (build < 0 || build > (int)maxValue)
-                    {
-                        //alink would generate an error here saying "Cannot auto-generate build and 
-                        //revision version numbers for dates previous to January 1, 2000." Without
-                        //some refactoring here to relay the date problem, Roslyn
-                        //will generate an inaccurate error about the version string being of the wrong format.
-
-                        version = null;
-                        return false;
-                    }
-                    values[2] = (ushort)build;
+                    values[i] = ushort.MaxValue;
```

Comment:
This behavior should probably be reflected in the doc comment for this method and probably in the doc comment of a public method that calls this helper.